src/backend/utils/init/miscinit.c

```diff
@@ -1688,6 +1688,17 @@ process_shared_preload_libraries(void)
 {
 	process_shared_preload_libraries_in_progress = true;
 
+	if ((shared_preload_libraries_string == NULL || strlen(shared_preload_libraries_string) == 0)
+		&& (IsBootstrapProcessingMode() || IsInitProcessingMode()))
+	{
+		/*
+		 * if we are in init or bootstrap mode. postgresql.conf does not exists yet.
+		 * try to load libraries from os environment
+		 */
+		char * const env_preload = getenv("GP_PRELOAD_LIBRARIES");
```

Comment:
What's the difference between `GP_PRELOAD_LIBRARIES` and `PRELOAD_LIBRARIES`?

Reply:
`GP_PRELOAD_LIBRARIES` is an environment variable, `PRELOAD_LIBRARIES` is a shell variable (but pass in to create_segment.sh using environment)